tasks/install_fpga_packages.yml

```diff
@@ -0,0 +1,143 @@
+    - name: check fpga driver package directory exist
+      stat:
+        path: "{{ fpga_rsu_driver_dir }}"
+      register: fpga_driver_dir_result
+
+    - name: Create fpga driver package directory if not exist
+      file:
+        path: "{{ fpga_rsu_driver_dir }}"
+        state: directory
+        mode: 0755
+      when: not (fpga_driver_dir_result.stat.exists | bool)
+
+- name: Download and untar FPGA driver tarball
+  block:
+    - name: Download fpga driver tarball
+      get_url:
+        url: "{{ fpga_rsu_driver_pkg_url }}"
+        dest: "{{ fpga_rsu_driver_dir }}/{{ fpga_rsu_tarball_name }}"
+      register: fpga_driver_tarball_downloaded
+      failed_when: false
+
+    - name: Fail if fpga driver tarball download fail.
+      fail:
+        msg: "FPGA Driver tarball download failed. Please check correct tarball URL is provided."
+      when:
+        - fpga_driver_tarball_downloaded is failed
+
+    - name: untar fpga driver tarball
+      unarchive:
+        src: "{{ fpga_rsu_driver_dir }}/{{ fpga_rsu_tarball_name }}"
+        dest: "{{ fpga_rsu_driver_dir }}"
+      register: fpga_driver_tarball_untar
+      failed_when: false
+
+- name: Setting fact and get rpm list
+  block:
+    - name: setting fact for fpga driver rpm directory
+      set_fact:
+        fpga_driver_rpm_dir: "{{ fpga_rsu_driver_dir }}/{{ fpga_rsu_tarball_name | splitext | first | splitext | first }}"
+
+    - name: get rpm list for installation
+      shell: ls -1 *.rpm
+      args:
+        chdir: "{{ fpga_driver_rpm_dir }}/opae"
+      register: opae_list_of_rpms
+      when: fpga_driver_tarball_untar is not failed
+
+    - name: get version of opae package
+      shell: rpm -qp --queryformat '%{VERSION}-%{RELEASE}' "opae-libs*"
+      args:
+        chdir: "{{ fpga_driver_rpm_dir }}/opae"
+      register: opae_package_version
+      when: opae_list_of_rpms.stdout_lines | length > 0
+
+- name: Install packages and verify installation
+  block:
+    - name: install fpga drivers
+      shell:
+        rpm -ivh {{ opae_list_of_rpms.stdout_lines | join(' ') }}
+      args:
+        chdir: "{{ fpga_driver_rpm_dir }}/opae"
+      register: opae_packages_installed
+
+    - name: Fail when OPAE FPGA package installation failed
+      fail:
+        msg: "OPAE FPGA package installation failed."
+      when: opae_packages_installed.rc != 0
+
+    - name: Verify opae and intel packages installed
+      shell: |-
+        set -o pipefail
+        rpm -qa | grep 'opae\|intel'
+      register: check_opae_packages_installed
+      when: opae_packages_installed.rc == 0
+
+    - name: Check installed packages version
+      shell:
+        rpm -q --queryformat '%{VERSION}-%{RELEASE}' "opae-libs"
+      register: opae_installed_version
+
+    - name: Fail when rpm package version and installed version mismatch.
+      fail:
+        msg: "Specified RPM package version and installed version mismatch."
+      when: opae_package_version.stdout != opae_installed_version.stdout
```

Comment:
...those checks. IMHO `rpm --verify` is checking more and you don't need to register and handle new variables.